test/jdk/java/net/Socket/CloseAvailable.java

```diff
@@ -48,110 +48,135 @@ public static void main(String[] args) throws Exception {
         testIOEOnClosed(false);
     }
 
+    /*
+     * Verifies that the Socket.getInputStream().available() throws an IOException
+     * if invoked after the socket has been closed.
+     */
     static void testClose() throws IOException {
-        boolean error = true;
-        InetAddress addr = InetAddress.getLocalHost();
-        ServerSocket ss = new ServerSocket(0, 0, addr);
-        int port = ss.getLocalPort();
-
-        Thread t = new Thread(new Thread("Close-Available-1") {
-            public void run() {
-                try {
-                    Socket s = new Socket(addr, port);
-                    s.close();
-                } catch (Exception e) {
-                    e.printStackTrace();
-                }
-            }
-        });
-
-        t.start();
-
-        Socket  soc = ss.accept();
-        ss.close();
-
-        DataInputStream is = new DataInputStream(soc.getInputStream());
-        is.close();
-
+        System.out.println("testClose");
+        final InetAddress addr = InetAddress.getLoopbackAddress();
+        final Socket acceptedSocket;
+        try (final ServerSocket ss = new ServerSocket(0, 0, addr)) {
+            System.out.println("created server socket: " + ss);
+            final int port = ss.getLocalPort();
+            // start a thread which initiates a socket connection to the server
+            Thread.ofPlatform().name("Close-Available-1")
+                    .start(() -> {
+                        try {
+                            final Socket s = new Socket(addr, port);
+                            System.out.println("created socket: " + s);
+                            s.close();
+                            System.out.println("closed socket: " + s);
+                        } catch (Exception e) {
+                            System.err.println("exception in " + Thread.currentThread().getName()
+                                    + ": " + e);
+                            e.printStackTrace();
+                        }
+                    });
+            // accept the client connect
+            acceptedSocket = ss.accept();
+            System.out.println(ss + " accepted connection " + acceptedSocket);
+        } // (intentionally) close the ServerSocket
+
+        final DataInputStream is = new DataInputStream(acceptedSocket.getInputStream());
+        is.close(); // close the inputstream and thus the underlying socket
+        System.out.println("closed inputstream of socket: " + acceptedSocket);
         try {
-            is.available();
-        }
-        catch (IOException ex) {
-            error = false;
+            final int av = is.available();
+            // available() was expected to fail but didn't
+            throw new AssertionError("Socket.getInputStream().available() was expected to fail on "
```

Comment:
Throwing an assertion error directly here is much better then the former fall-through approach.